src/Markdown.MAML/Renderer/Markdownv2Renderer.cs

```diff
@@ -608,5 +615,19 @@ public static string GetEscapedMarkdownText(string text)
 
                 ;
         }
+
+        private string DetectLanguage(string hint)
+        {
+            // Detect PowerShell based on first line
+            // - Look PS C:\> prefix followed 
+            // - Look for standalone PowerShell verbs-noun, current verbs use 3-11 characters
+            // - Look for inline help # character
```

Comment:
1) Let's move this comments closer to the regex.
2) "Look for inline help # character" - I don't think it's a good idea
3) Look for standalone PowerShell verbs-noun will probably match everything

Reply:
@vors Fair calls. I think # and verb-noun in the context of the first line of code blocks that haven't been marked as another language is fine, and is quite unlikely to match output blocks.

But I will defer to you if that's your recommendation. :) The case that is particularly handy for the verb noun is where comment based help engine strips it out.